src/baldr/graphreader.cc

```diff
@@ -340,13 +340,18 @@ TileCache* TileCacheFactory::createTileCache(const boost::property_tree::ptree&
 }
 
 // Constructor using separate tile files
-GraphReader::GraphReader(const boost::property_tree::ptree& pt)
+GraphReader::GraphReader(const boost::property_tree::ptree& pt,
+                         std::unique_ptr<tile_getter_t>&& tile_getter)
     : tile_extract_(get_extract_instance(pt)), tile_dir_(pt.get<std::string>("tile_dir", "")),
-      curlers_(std::make_unique<curler_pool_t>(pt.get<size_t>("max_concurrent_reader_users", 1),
-                                               pt.get<std::string>("user_agent", ""))),
-      tile_url_(pt.get<std::string>("tile_url", "")),
-      tile_url_gz_(pt.get<bool>("tile_url_gz", false)),
-      cache_(TileCacheFactory::createTileCache(pt)) {
+      tile_getter_(std::move(tile_getter)),
+      max_concurrent_users_(pt.get<size_t>("max_concurrent_reader_users", 1)),
+      tile_url_(pt.get<std::string>("tile_url", "")), cache_(TileCacheFactory::createTileCache(pt)) {
+  if (!tile_getter_) {
```

Comment:
shouldn't this be ` if (!tile_getter_ && !tile_url_.empty()) {` surely we dont want to initialize the `tile_getter_` when the optional url in the config file wasn't specified. it doesnt make sense for the graph reader to try to get tiles from URL if the person explicitly didnt configure the graphreader to hit a url

Reply:
that's a good catch and I didn't know it! thanks for noticing, will fix it.

Reply:
added this condition to if